// instatus.ts/lib/metrics/MetricManager.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
/* eslint-disable @typescript-eslint/restrict-template-expressions */
var __1 = require("..");
var Metric_1 = __importDefault(require("./Metric"));
/**
 * A manager of the metrics
 *
 * @export
 * @class MetricManager
 * @extends {Base}
 */
var MetricManager = /** @class */ (function (_super) {
    __extends(MetricManager, _super);
    function MetricManager() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    // GET /v1/:page_id/metrics
    /**
     * Gets all metrics
     *
     * @return {*}  {Promise<Metric[]>}
     * @memberof MetricManager
     */
    MetricManager.prototype.getAll = function () {
        return __awaiter(this, void 0, void 0, function () {
            var metrics;
            var _this = this;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this._check();
                        return [4 /*yield*/, this.request('get', this.client.pageID + "/metrics")];
                    case 1:
                        metrics = _a.sent();
                        return [2 /*return*/, metrics.map(function (m) { return new Metric_1.default(_this.client, m); })];
                }
            });
        });
    };
    // GET /v1/:page_id/metrics/:metric_id
    /**
     * Gets one metric
     *
     * @param {string} metricID The metric ID
     * @return {*}  {Promise<Metric>}
     * @memberof MetricManager
     */
    MetricManager.prototype.get = function (metricID) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this._check(metricID);
                        return [4 /*yield*/, this.request('get', this.client.pageID + "/metrics/" + metricID)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // PUT /v1/:page_id/metrics/:metric_id
    /**
     * Updates a metric
     *
     * @param {string} metricID The metric ID
     * @param {MetricPut} data The data to update
     * @return {*}  {Promise<Metric>}
     * @memberof MetricManager
     */
    MetricManager.prototype.update = function (metricID, data) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this._check(metricID);
                        return [4 /*yield*/, this.request('put', this.client.pageID + "/metrics/" + metricID, data)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // DELETE /v1/:page_id/metrics/:metric_id
    /**
     * Deletes a metric
     *
     * @param {string} metricID The metric ID
     * @return {*}  {Promise<Metric>}
     * @memberof MetricManager
     */
    MetricManager.prototype.delete = function (metricID) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this._check(metricID);
                        return [4 /*yield*/, this.request('delete', this.client.pageID + "/metrics/" + metricID)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // POST /v1/:page_id/metrics/:metric_id
    /**
     * Adds a datapoint to a metric
     *
     * @param {string} metricID The metric ID
     * @param {MetricDataPointPost} data The data to add
     * @return {*}  {Promise<MetricDataPoint>}
     * @memberof MetricManager
     */
    MetricManager.prototype.add = function (metricID, data) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this._check();
                        return [4 /*yield*/, this.request('post', this.client.pageID + "/metrics/" + metricID, data)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    // POST /v1/:page_id/metrics/:metric_id/data
    /**
     * Adds multiple data to a metric
     *
     * @param {string} metricID The metric ID
     * @param {MetricDataPointPost[]} data The data to add
     * @return {*}  {Promise<MetricDataPointPost[]>}
     * @memberof MetricManager
     */
    MetricManager.prototype.adds = function (metricID, data) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this._check();
                        return [4 /*yield*/, this.request('post', this.client.pageID + "/metrics/" + metricID + "/data", data)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    MetricManager.prototype._check = function (metricID) {
        if (metricID !== undefined && typeof metricID !== 'string')
            throw Error('ComponentID must be a string');
        if (this.client.pageID === undefined)
            throw Error('PageID undefined');
    };
    return MetricManager;
}(__1.Base));
exports.default = MetricManager;
